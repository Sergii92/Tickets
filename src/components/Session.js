import React, { useState } from "react";
import SessionPage from "../pages/SessionPage";
import { SessionTitle, SessionBlock } from "../styled/components";
import { history } from "../constants/history";

const Session = ({ sesssion }) => {
  const [modalIsOpen, setModalIsOpen] = useState(false);

  const openModal = (e) => {
    history.push(`/session`);
    setModalIsOpen(!modalIsOpen);
    localStorage.setItem("modal", sesssion.name);
  };

  const closeModal = () => {
    setModalIsOpen(!modalIsOpen);

    history.push(`/home`);
    localStorage.clear();
  };

  return (
    <>
      <SessionBlock onClick={openModal}>
        <SessionTitle>{sesssion.time}</SessionTitle>
        <SessionTitle>{sesssion.name}</SessionTitle>
        <SessionTitle>
          Free places:{sesssion.totalPlaces - sesssion.booked}
        </SessionTitle>
        <SessionTitle>Booked places:{sesssion.booked}</SessionTitle>
      </SessionBlock>
      <SessionPage
        sesssion={sesssion}
        modalIsOpen={modalIsOpen}
        setModalIsOpen={setModalIsOpen}
        closeModal={closeModal}
      />
    </>
  );
};

export default Session;
