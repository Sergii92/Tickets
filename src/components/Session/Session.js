import React from 'react'
import styled from 'styled-components';

import img from '../../assets/img/img.png'


const SessionWrapper = styled.div`
border: 0px solid rgba(0, 0, 0, 0);
border-radius: 5px;
width:33%;
max-width:300px;
border:1px solid black;
padding: 20px;
text-align:center;
background-color:black;
color:white


`
const SessionImageBlock = styled.div`
width:95%;
background-image:url(${img});
background-position:center;
background-repeat:no-repeat;

border-radius:10px;
height:300px;
margin:0 auto;
background-color:#9d9d9d;
margin-bottom:10px;


`


 const Session = () => {
  return (
  <SessionWrapper> 
    <SessionImageBlock/> 
   <h2>SessionName</h2>
  <h3>SessionTime</h3>

  </SessionWrapper>
  )
}


export default Session