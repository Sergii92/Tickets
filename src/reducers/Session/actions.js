import { TYPES } from "./types";

export const dataActions = {
  fetchData: () => ({
    type: TYPES.FETCH_DAYS,
  }),
  fetchDataSuccess: (data) => ({
    type: TYPES.FETCH_DATA_SUCCESS,
    payload: data,
  }),
  getSessionID: (id) => ({
    type: TYPES.GET_SESSION_ID,
    payload: id,
  }),
  fetchSession: () => ({
    type: TYPES.FETCH_SESSIONS,
  }),
  fetchSessionsSuccess: (data) => ({
    type: TYPES.FETCH_SESSIONS_SUCCESS,
    payload: data,
  }),

  fetchPlacesSuccess: (data) => ({
    type: TYPES.FETCH_PLACES,
    payload: data,
  }),
};
