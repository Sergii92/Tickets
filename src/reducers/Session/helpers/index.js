export const helperBoockedPlace = (obj, arr, id) => {
  const newPlaces = arr.map((place) => {
    if (place.id === id) {
      return {
        ...place,
        boocked: true,
      };
    } else {
      return {
        ...place,
      };
    }
  });

  return {
    ...obj,
    places: newPlaces,
  };
};

export const helperGetPlaces = (obj, arr) => {
  return {
    ...obj,
    places: arr,
  };
};
