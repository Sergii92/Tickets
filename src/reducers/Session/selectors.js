export const selectDays = (state) => state.data.days;

export const selectSessions = (state) => state.data.sessions;
export const selectPlaces = (state) => state.data.places;

export const selectSessionID = (state) => state.data.sessionID;
