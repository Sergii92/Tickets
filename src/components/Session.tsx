import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { Card } from "react-bootstrap";

import { CustomCard } from "../styled/components";
import MyModal from "./Modal";

import { dataActions } from "../redux/ducks/session";
import { history } from "../constants/history";
import { IsessionProps } from "../interfaces";
import SingleCard from "./SingleCard";

const Session = ({ sesssion }: IsessionProps) => {
  const dispatch = useDispatch();

  const [show, setShow] = useState(false);

  const handleClose = () => {
    setShow(false);
    history.push(`/home`);
  };
  const handleShow = (id: number) => {
    setShow(true);

    dispatch(dataActions.fetchPlaces());
    history.push(`/sesssion${id}`);
  };

  return (
    <>
      <SingleCard
        id={sesssion.id}
        sessionName={sesssion.sessionName}
        time={sesssion.time}
        onClick={() => handleShow(sesssion.id)}
      />
      <MyModal handleClose={handleClose} show={show} sesssion={sesssion} />
    </>
  );
};

export default Session;
