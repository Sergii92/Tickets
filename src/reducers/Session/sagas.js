import { apply, call, put, takeEvery } from "redux-saga/effects";

import { TYPES } from "./types";
import { dataActions } from "./actions";

const fetchDay = () => {
  return fetch("http://localhost:3004/days");
};

const fetchSessionUrl = () => {
  return fetch(`http://localhost:3004/sessions`);
};

const fetchPlacesUrl = () => {
  return fetch(`http://localhost:3004/ticket`);
};

const update = ({ payload }) => {
  return fetch(`http://localhost:3004/ticket/1000`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },

    body: JSON.stringify({ places: payload }),
  });
};

export function* fetchSessionsWorker({ payload }) {
  try {
    const responce = yield call(() => fetchSessionUrl(payload));

    const data = yield apply(responce, responce.json);
    if (responce.status !== 200) {
      throw new Error("error");
    }
    yield put(dataActions.fetchSessionsSuccess(data));
  } catch (error) {
    console.log(error);
  }
}

export function* fetchDays() {
  try {
    const responce = yield call(fetchDay);
    const data = yield apply(responce, responce.json);
    if (responce.status !== 200) {
      throw new Error("error");
    }
    yield put(dataActions.fetchDataSuccess(data));
  } catch (error) {
    console.log(error);
  }
}
export function* fetchPlacesWorker() {
  try {
    const responce = yield call(fetchPlacesUrl);
    const data = yield apply(responce, responce.json);
    if (responce.status !== 200) {
      throw new Error("error");
    }
    yield put(dataActions.fetchPlacesSuccess(data[0].places));
  } catch (error) {
    console.log(error);
  }
}

export function* sagaWatcher() {
  yield takeEvery(TYPES.FETCH_DAYS, fetchDays);
  yield takeEvery(TYPES.FETCH_SESSIONS, fetchSessionsWorker);
  yield takeEvery(TYPES.FETCH_PLACES, fetchPlacesWorker);
  yield takeEvery(TYPES.PUT_PLACES, update);
}
