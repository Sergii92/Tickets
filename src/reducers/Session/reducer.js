import { TYPES } from "./types";
import { helperBoockedPlace, helperGetCarentPlaces } from "./helpers/index";

const initialState = {
  data: [],
  places: [],
};

export const dataReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case TYPES.FETCH_DATA_OK:
      return {
        ...state,
        data: payload,
      };
    case TYPES.BOOK_PLACE:
      return {
        ...state,
        data: helperBoockedPlace(state.data, payload),
      };
    case TYPES.GET_CARENT_PLACES:
      return {
        ...state,
        places: helperGetCarentPlaces(state.data, payload),
      };
    default:
      return state;
  }
};
