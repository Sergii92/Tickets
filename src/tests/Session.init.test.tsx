import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Session from "../components/Session";

const mockStore = configureStore();

describe("Session  Component", () => {
  const store = mockStore({});
  const sessionProps = {
    daysId: 1,
    time: "10:00",
    id: 11,
    description:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam possimus quaerat ipsa dolor nihil cum ullam iste culpa veniam, magni a neque beatae consequatur totam eaque maxime aperiam illum molestiae.",
    sessionName: "Sunday session1",
  };

  describe("Session component render", () => {
    it("RENDER", () => {
      const component = shallow(
        <Provider store={store}>
          <Session sesssion={sessionProps}></Session>
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });
    it("Find text", () => {
      const component = shallow(
        <Provider store={store}>
          <Session sesssion={sessionProps}></Session>
        </Provider>
      );

      // const node = component.find("p");

      // expect(node.length).toEqual(2);
    });
  });

  describe("detailse", () => {
    let mySession: any;
    let component: any;
    beforeEach(() => {
      mySession = {
        daysId: 1,
        time: "10:00",
        id: 11,
        description:
          "Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam possimus quaerat ipsa dolor nihil cum ullam iste culpa veniam, magni a neque beatae consequatur totam eaque maxime aperiam illum molestiae.",
        sessionName: "Sunday session1",
      };
    });

    beforeEach(() => {
      component = shallow(
        <Provider store={store}>
          <Session sesssion={mySession}></Session>
        </Provider>
      );
    });

    it("should exist", () => {
      expect(component).toBeTruthy();
      expect(component.props().children.props.sesssion).toEqual(mySession);
    });
  });
});
