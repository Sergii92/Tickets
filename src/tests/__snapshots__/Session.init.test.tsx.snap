// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session  Component Session component render RENDER 1`] = `ShallowWrapper {}`;
